actions/npm-basic-checks/action.yml

```diff
@@ -0,0 +1,26 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of Node.
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: "Basic Checks and Tests"
+description: "Perform basic checks and tests for an NPM project"
+inputs:
+  node-version:
+    required: true
+    description: "the node version"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ inputs.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
+    - run: npm install
+      shell: bash
+    - run: npx prettier --check .
+      shell: bash
+    - run: npm run lint
+      shell: bash
+    - run: npm run test
```

Comment:
I like the idea how it will enforce every repo to implement at least lint and test scripts but I also would like them to be separate (parallel) jobs in github actions terms so they run as 3 checks and not one to actually report all issues in one run rather than fail on first failing step. I'm not sure how that would work with composite actions though but one workaround would be to define `matrix` of checks since it would run them all in parallel.

Reply:
imo lint should run first, then all tests can run in parallel, then any release etc.

If the linting is failing, we should just quit early. I don't think there is a benefit to running the lint and tests in parallel with how quickly linting typically is.

Reply:
i think this also could be a follow up as it involves more work with our other repos.

Reply:
I looked into it and you can't have parallel steps inside a composite action, because a composite action doesn't define the environment at all. It doesn't even have its own shell defined which is why you need the `shell: bash` lines. So you can only define e.g. a matrix in the action that calls the composite action.

So I could make a separate composite action for each step e.g. `npm run lint`, `npm run test` etc. but at that point of granularity, why have the composite action in the first place? It's more trouble than its worth.

I would agree that these checks should run in sequence. This action is meant to be a "basic checks" so it's not really applicable to more complicated CI workflows like e.g. `skynet-js`, but sufficient for most of our other repos.